utils/evaluate.py

```diff
@@ -28,6 +29,79 @@ def get_single_dcg_for_rankers(descending_rankings, document_labels, max_len):
                   + 2)[None, :], axis=1)
 
 
+# Calculate NDCG with a single array of descending ranking and the corresponding labels.
+def get_ndcg_with_labels(ranking, labels, max_len, initial, is_test = False, impression = -1):
+
+    # if is_test and initial == 280:
+    #     print labels
+
+    idcg = get_idcg(np.asarray(labels), max_len)
+    if idcg==0:
+        return 0
+    nominators = [2. ** label - 1. for label in labels]
+    denominators = [math.log(r+2., 2) for r in ranking]
+    for i in range(len(ranking)):
+        if ranking[i]>=max_len:
+            nominators[i] = 0
+
+    ndcg = 0
+    for i in range(len(nominators)):
+        ndcg += (nominators[i] / denominators[i]) / idcg
+
+    if impression % 100 == 0:
+        print ndcg
+    return ndcg
+
+# Calculate NDCG with a single array of descending model_ranking and a descending ideal_ranking.
+# Assume that the first "max_len" documents in ideal_ranking has relevance score 1 and other documents
+# have relevance score 0.
+def get_ndcg_with_ranking(model_ranking, ideal_ranking, max_len, initial):
+    labels = [0 for i in range(len(model_ranking))]
+    displayed_ideal_ranking = ideal_ranking[:max_len]
+    for document in model_ranking:
+      if document in displayed_ideal_ranking and document<len(model_ranking):
+          labels[document] = 1
+
+    return get_ndcg_with_labels(model_ranking, labels, max_len, initial)
+
+def get_ndcg_with_ranking2(model_ranking, ideal_ranking, max_len, initial, impression):
```

Comment:
What's the difference between get_ndcg_with_ranking and get_ndcg_with_ranking2 ? Which one should we use?